### PR TITLE
Harry Potter Quiz Game

### LINKED ISSUE #5133: [New game]: Harry Potter wizard quiz game.
### 🎮 Game Request

Game logic and basic description
The Harry Potter Wizard Quiz Game is a trivia-based game designed for fans of the Harry Potter series. Players test their knowledge of the magical world, answering questions about characters, spells, potions, and key events from the books and movies. Points are earned for correct answers, and the goal is to achieve the highest score, proving your expertise as a true wizard or witch.

### Point down the features

Game points

Diverse Question Categories: Includes questions on characters, spells, potions, magical creatures, Hogwarts houses, and key events from both the books and movies.

Time-Based Challenges: Players must answer questions within a set time limit to earn points, adding an element of urgency to the gameplay.

Interactive Animations and Sound Effects: Engaging visual and audio effects that bring the magical world to life with every correct or incorrect answer.

### Select program in which you are contributing

GSSoC24

### Code of Conduct

- [X] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) of this project.

### PR DESCRIPTION
**Pull request by sweta-singh28**
## PR Description 📜
 Added a new game.

Fixes #5133 

<hr>
 
## Mark the task you have completed ✅

- [x] I follow [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md) & [CODE OF CONDUCT](https://github.com/kunjgit/GameZone/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I have followed proper naming convention showed in [CONTRIBUTING GUIDELINE](https://github.com/kunjgit/GameZone/blob/main/.github/CONTRIBUTING_GUIDELINE.md)
- [x] I have added screenshot for website preview in assets/images 
- [x] I have added entries for my game in GameZone's README.md
- [x] I have added working video of the game in README.md (optional)
- [x] I have specified the respective issue number for which I have requested the new game.
- [x] I have added my Game to gamesData.json file and it is loading properly on main website as intended.

<hr>

## Add your screenshots(Optional) 📸
Screenshot 2024-08-09 145254

Screenshot 2024-08-09 141828

Screenshot 2024-08-09 144901

https://github.com/user-attachments/assets/8c478898-221f-42d7-8475-b24837c1eadd

--- 
<br>

## Thank you soo much for contributing to our repository 💗

### CODE REVIEW

**Comment by kunjgit** [comment]
Thank you @sweta-singh28 ,for creating the PR and contributing to our GameZone 💗 
 Review team will review the PR and will reach out to you soon! 😇 
 Make sure that you have marked all the tasks that you are done with ✅. 
 Thank you for your patience! 😀

**Comment by sweta-singh28** [comment]
@kunjgit sir, done.

File: assets/js/index.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 const generateLiTags = (gamesData, searchText = "") => {
   const liTags = [];
   searchText = searchText.trim().toLowerCase(); // Trim whitespace and convert to lowercase
-  let lengthOfJson = Object.keys(gamesData).length; // get the no. of games of Json to list them all
+  let lengthOfJson = Object.keys(gamesData).length+10; // get the no. of games of Json to list them all
```
Comment (ishita-43) [inline assets/js/index.js:6]: Remove 100 from the end if not needed

**Review by ishita-43 — CHANGES_REQUESTED** [review]
Add your game entry in main readme. Md file of the repo refer other entries for the format

File: assets/js/index.js (hunk 1 above)
Comment (sweta-singh28) [reply assets/js/index.js:6]: @ishita-43 mam,  It is needed because due to some unknown issue games after 639 are not getting rendered.

Comment (ishita-43) [reply assets/js/index.js:6]: Ok, if this way it works fine we will approve

**Comment by sweta-singh28** [comment]
@ishita-43 maam, done.

File: assets/js/index.js (hunk 1 above)
Comment (sweta-singh28) [reply assets/js/index.js:6]: done

**Review by ishita-43 — APPROVED** [review]

**Comment by sweta-singh28** [comment]
@kunjgit sir,  Could you please merge this PR asap cause Today is the last day for GSSoC contributions.

**Comment by kunjgit** [comment]
Thank you  @sweta-singh28 , for your valuable time and contribution in our GameZone 💗. 
 It’s our GameZone, so Let’s build this GameZone altogether !!🤝
  Hoping to see you soon with another PR again 😇 
 Wishing you all the best for your journey into Open Source🚀

**Pull request merged**